crates/lib/src/fetchers/gh_crate_meta/hosting.rs

```diff
@@ -0,0 +1,40 @@
+use url::Url;
+
+use crate::errors::BinstallError;
+
+#[derive(Copy, Clone, Debug)]
+pub enum GitHostingServices {
+    GitHub,
+    GitLab,
+    BitBucket,
+    SourceForge,
+    Unknown,
+}
+impl GitHostingServices {
+    pub fn guess_git_hosting_services(repo: &Url) -> Result<Self, BinstallError> {
+        use GitHostingServices::*;
+
+        match repo.domain() {
+            Some(domain) if domain.starts_with("github") => Ok(GitHub),
+            Some(domain) if domain.starts_with("gitlab") => Ok(GitLab),
```

Comment:
We also have a few people using `www.github.com`, `www.gitlab.com`... maybe we could strip the `www.` for matching purposes?

Reply:
I called `get_redirected_final_url` on the repo url before calling this function so that `www.github.com` will be resolved to `github.com`